index.bs

```diff
@@ -769,63 +809,84 @@ spec:css-syntax-3;
   <ol class="algorithm">
     1.  Let |settings| be the <a>current settings object</a>
 
-    2.  Assert: |settings| is a [=secure context=].
+    1.  Assert: |settings| is a [=secure context=].
 
-    3.  If <code>|options|.{{CredentialRequestOptions/signal}}</code>'s [=AbortSignal/aborted flag=]
+    1.  If <code>|options|.{{CredentialRequestOptions/signal}}</code>'s [=AbortSignal/aborted flag=]
         is set, then return [=a promise rejected with=] an "{{AbortError}}" {{DOMException}}.
 
-    4.  Let |p| be [=a new promise=].
+    1.  If <code>|options|.{{CredentialRequestOptions/mediation}}</code> is
+        "{{CredentialMediationRequirement/conditional}}":
+
+        1.  For each |interface| in |options|' [=relevant credential interface objects=]:
 
-    5.  Let |origin| be the [=current settings object=]'s [=environment settings object/origin=].
+            1.  If |interface| does not support {{CredentialMediationRequirement/conditional}}
+                [=user mediation=], return a "{{TypeError}}" {{DOMException}}.
```

Comment:
Because every problem and non-problem in computer science is solved by another layer of indirection:

```suggestion
                [=user mediation=], return [=a promise rejected with=] a "{{TypeError}}" {{DOMException}}.
```